Make wasm plugin work with service workers

vedantroy opened the pull request:
The code generated by @rollup/plugin-wasm breaks when executed in a
service worker because window is undefined.

<!--
  ⚡️ katchow! We ❤️ Pull Requests!

  If you remove or skip this template, you'll make the 🐼 sad and the mighty god
  of Github will appear and pile-drive the close button from a great height
  while making animal noises.

  Pull Request Requirements:
  * Please include tests to illustrate the problem this PR resolves.
  * Please lint your changes by running `npm run lint` before creating a PR.
  * Please update the documentation in `/docs` where necessary

  Please place an x (no spaces - [x]) in all [ ] that apply.
-->

<!-- the plugin(s) this PR is for -->

## Rollup Plugin Name: `wasm`

This PR contains:

- [X] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [X] yes (_bugfixes and features will not be merged without tests_)
- [] no

~~This is a draft PR because I'm not sure how to write a test that runs code inside a service worker. Once I figure that out, I will add a test and update the above checkbox.~~

Breaking Changes?

- [x] yes (_breaking changes will not be merged unless absolutely necessary_)
- [ ] no

List any relevant issue numbers: #333 

### Description

<!--
  Please be thorough and clearly explain the problem being solved.
  * If this PR adds a feature, look for previous discussion on the feature by searching the issues first.
  * Is this PR related to an issue?
-->

~~This PR adds a check for whether `window` is undefined.~~ This PR makes the loader function use `globalThis` instead of `window`. Prior to this, `@rollup/plugin-wasm` would break when the resulting generated code was run in a service worker, because it assumed `window` was defined.

shellscape commented:
Thanks for taking this on. We'll leave this open and pending until you give the go-ahead to review.

vedantroy commented:
@shellscape I added a test and refactored test.js because I needed to pass the code generated by rollup into a `Worker` instead of `testBundle`.

Note: The test is a bit fragile because I have to fake the browser environment (`atob` does not exist on `globalThis` in node, so I polyfilled it, and I undefined `process` to force the wasm loader to go into the browser section). However, I think it's the most reasonable test that is possible.

shellscape commented:
Thanks for the PR!

@ColinEberhardt and @jamen would you two be willing to take a look at this one?

vedantroy commented:
@jamen The gap of support seems to show that WebAssembly is less supported than `globalThis` in all cases (except for Opera Mobile, which supports wasm but not `globalThis`). However, if this is an issue, I can add a check so it tries using `window` before `globalThis`.

jamen commented:
What I mean is the browser versions. WebAssembly came first and globalThis came later, so there are versions where WebAssembly exists but globalThis doesn't. I believe that is a breaking change, because the build would produce a more limited range of support than WebAssembly itself.

I don't think its an issue, but I think it should be in a major version. I haven't been very involved anymore though, so someone else can make that call. But also, a build concerned about this would be using something like `babel-preset-env` anyways.

shellscape commented:
Thanks everyone!
